src/AutoRest.CSharp/Common/Output/Models/Types/ExpressionTypeProvider.cs

```diff
@@ -12,29 +13,52 @@ namespace AutoRest.CSharp.Output.Models.Types
     // TODO -- eventually we should combine everything in this class into TypeProvider
     internal abstract class ExpressionTypeProvider : TypeProvider
     {
-        protected ExpressionTypeProvider(string defaultNamespace, SourceInputModel? sourceInputModel) : base(defaultNamespace, sourceInputModel)
+        protected ExpressionTypeProvider(string defaultNamespace, SourceInputModel? sourceInputModel)
+            : base(defaultNamespace, sourceInputModel)
         {
+            DeclarationModifiers = ClassSignatureModifiers.Partial | ClassSignatureModifiers.Public;
         }
 
         private IReadOnlyList<string>? _usings;
         public IReadOnlyList<string> Usings => _usings ??= BuildUsings().ToArray();
 
+        public ClassSignatureModifiers DeclarationModifiers { get; init; }
+
         protected virtual IEnumerable<string> BuildUsings()
         {
             yield break;
         }
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override string DefaultAccessibility { get; } = "public";
+
         public virtual CSharpType? Inherits { get; protected init; }
 
         private IReadOnlyList<CSharpType>? _implements;
         public virtual IReadOnlyList<CSharpType> Implements => _implements ??= BuildImplements().ToArray();
 
+        private IReadOnlyList<PropertyDeclaration>? _properties;
+        public virtual IReadOnlyList<PropertyDeclaration> Properties => _properties ??= BuildProperties().ToArray();
```

Comment:
I think in the near future we might need to let `PropertyDeclaration` and `FieldDeclaration` to inherit the same base type such as `MemberDeclaration` because in a lot of cases properties and fields are alike and could be replaced with each other - for instance the `AdditionalProperties` property and the `_serializedAdditionalRawData` field - one exists and the other disappears
I will create an issue for this, and we might need a discussion on how we define a TypeProvider when migrating these to the new generator

Reply:
Issue here: https://github.com/Azure/autorest.csharp/issues/4312

Reply:
I prefer to keep these separate.  Right now field declaration has a lot of stuff like write as property which we will remove in the new generator.